src/model_builder.cpp

```diff
@@ -104,7 +108,15 @@ auto make_primitive_visualization_geometry(const Collision_primitive_type type,
                                         cube_vertex_normals, cube_vertex_texcoords),
                                std::make_tuple(glm::vec3{size}));
       default:
-         std::terminate(); //
+//TODO: FIX MESSY
+#ifdef _WIN32
+        std::terminate(); //Not sure how to make this portable, terminates entire process
+                          //on Mac/Linux. Will look deeper into it someday...
+#else
+        return std::tuple_cat(as_spans(cube_indices, cube_vertex_positions,
```

Comment:
There are a number of changes like this one, I don't know of a good way to terminate individual threads w/POSIX.